tests/yam/migration/migration_unattended.pm

```diff
@@ -20,15 +21,27 @@ sub run {
 
     assert_script_run("echo 'url: " . get_var('SCC_URL') . "' > /etc/SUSEConnect");
 
-    my $repo_home = "http://download.suse.de/ibs/home:/fcrozat:/SLES16/SLE_\$releasever";
-    my $repo_images = 'http://download.suse.de/ibs/home:/fcrozat:/SLES16/images/';
+    my $repo_server = get_var('REPO_MIRROR_HOST', 'download.suse.de');
+    my $repo_home = "http://" . $repo_server . "/ibs/home:/fcrozat:/SLES16/SLE_\$releasever";
+    my $repo_images = 'http://' . $repo_server . '/ibs/home:/fcrozat:/SLES16/images/';
     zypper_call("ar --refresh -p 90 '$repo_home' home_sles16");
     zypper_call("ar --refresh -p 90 $repo_images home_images");
 
     # install the migration image and active it
     zypper_call("--gpg-auto-import-keys -n in suse-migration-sle16-activation");
 
-    power_action('reboot', textmode => 1, keepconsole => 1, first_reboot => 1);
+    # Deacivate unsupported extensions before doing migration
+    if (get_var('SCC_SUBTRACTIONS') and script_run("SUSEConnect -s | grep -i \"\\\"status\\\":\\\"Registered\\\"\"") == 0) {
+        foreach (split(',', get_var('SCC_SUBTRACTIONS'))) {
+            my $extension = get_addon_fullname($_);
+            remove_suseconnect_product($extension) if (script_run("SUSEConnect --list-extensions  | grep -i \"Deactivate with:.*$extension\"") == 0);
+        }
+    }
+
+    # Upload logs to know system state before migration
+    upload_logs("/boot/grub2/grub.cfg", failok => 1);
+    upload_folders(folders => '/etc/zypp/repos.d/');
+    power_action('reboot', keepconsole => 1, first_reboot => 1);
```

Comment:
textmode => 1 is forced on purpose "for now" we needed for some of the migration using desktop. Please keep it that way.

Reply:
Done